<launch>
    <!-- Arguments -->
    <arg name="global_mapping" default="false" />
    <arg name="rviz" default="false" />

    <!-- FastDEM Elevation Mapping Node -->
    <node name="fastdem" pkg="fastdem_ros" type="fastdem_ros_node" output="screen">
        <rosparam file="$(find fastdem_ros)/config/default.yaml" command="load" subst_value="true" />

        <!-- Global mapping mode: larger map, fixed origin, slower publish rate -->
        <param name="mapping_mode" value="global" if="$(arg global_mapping)" />
        <param name="map/width" value="200.0" if="$(arg global_mapping)" />
        <param name="map/height" value="200.0" if="$(arg global_mapping)" />
        <param name="topics/publish_rate" value="1.0" if="$(arg global_mapping)" />
    </node>

    <!-- RViz (optional) -->
    <node pkg="rviz" type="rviz" name="rviz" if="$(arg rviz)"
          args="-d $(find fastdem_ros)/launch/rviz/fastdem_ros.rviz" />
</launch>
